src/Dependencies/Collections/SegmentedList`1.cs

```diff
@@ -153,8 +153,13 @@ public int Capacity
                     var oldSegmentCount = segments.Length;
                     var newSegmentCount = (value + SegmentedArrayHelper.GetSegmentSize<T>() - 1) >> SegmentedArrayHelper.GetSegmentShift<T>();
 
-                    // Grow the array of segments, if necessary
-                    Array.Resize(ref segments, newSegmentCount);
+                    if (newSegmentCount > segments.Length)
+                    {
+                        // Grow the array of segments, if necessary. Note that this array may end up having null entries
+                        // at it's end when this method completes. Minimally doubling the outer array size allows amortized
+                        // linear cost growth properties.
+                        Array.Resize(ref segments, Math.Max(newSegmentCount, segments.Length * 2));
```

Comment:
📝 I don't like this being on the general `Capacity.set` path, since it could penalize users who are setting a capacity specifically because they know the exact size of the resulting collection.

Reply:
That makes sense, let me move this to Grow

Reply:
Tried to move this to Grow and not modify the Capacity setter, but it ended up needing to duplicate pretty much all the logic in the Capacity setter. So, I created a helper method that both places can call into that takes in a flag to specify whether the doubling is desired for the segment array.